truss-chains/truss_chains/utils.py

```diff
@@ -128,6 +130,25 @@ def get_free_port() -> int:
         return port
 
 
+def override_chainlet_to_service_metadata(
+    chainlet_to_service: Mapping[str, definitions.ServiceDescriptor],
+) -> Mapping[str, definitions.ServiceDescriptor]:
+    # Override predict_urls in chainlet_to_service ServiceDescriptors if chainlet_url_map exists in dynamic config
+    chainlet_url_map_str = dynamic_config_resolver.get_dynamic_config_value(
+        definitions.DYNAMIC_CONFIG_CHAINLET_URL_MAP_KEY
+    )
+    if chainlet_url_map_str:
+        chainlet_url_map = json.loads(chainlet_url_map_str)
```

Comment:
would love to be explicit about the typing here - either via or comment or by defining some dataclass that we instantiate from json

Reply:
wondering if we want some type of nested structure that allows us to add a value to this mapping at some point

e.g.
{ "name": { "predic_url": ... } }

This will make it easy if we ever need to add other values that map to a chainlet

Reply:
not necessary to do, but this gives us a more forward-extensible design. If i'm understanding it correctly, this is right now { "chainle_name": "https....." }

Reply:
Synced with Raymond, Marius, Sid. We decided to go with something like: 
```
"dynamic_chainlet_config": {
   "chainlet_1_name": {
       "predict_url": "predict_url_for_chainlet_1",
       # allows for us to add more fields here if we need to in the future
   }
   "chainlet_2_name": {
       "predict_url": "predict_url_for_chainlet_2",
       # allows for us to add more fields here if we need to in the future
   }
}
```
This is extensible and usable should we need it in the future. Updated the PR to reflect this new data structure.